modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoveryCoordinatorFailureTest.java

```diff
@@ -282,17 +282,17 @@ public void awaitDrop() throws InterruptedException, IgniteCheckedException {
         /** {@inheritDoc} */
         @Override protected void writeToSocket(
             Socket sock,
-            TcpDiscoveryAbstractMessage msg,
             byte[] data,
             long timeout
         ) throws IOException {
-            if (isDrop(msg)) {
-                // Replace logic routine message with a stub to update last-sent-time to avoid segmentation on
-                // connRecoveryTimeout.
-                msg = new TcpDiscoveryConnectionCheckMessage(locNode);
-            }
 
-            super.writeToSocket(sock, msg, data, timeout);
+            // Replace logic routine message with a stub to update last-sent-time to avoid segmentation on
```

Comment:
@aleksey-kurinov I'm sure we don't need this failure-emulation method in `FailingDiscoverySpi` at all because it passes no message. The `msg` is always null. This `writeToSocket` works with `data`. The test works well without emulation a failure here.